Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/manager_app/view/plan_create_task_list/PlanTaskListFragment.java

```diff
@@ -2,43 +2,74 @@
+import android.support.v7.widget.helper.ItemTouchHelper;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 
 import javax.inject.Inject;
 
+import database.entities.PlanTaskTemplate;
 import database.entities.PlanTemplate;
 import database.repository.PlanTemplateRepository;
 import pg.autyzm.friendly_plans.ActivityProperties;
 import pg.autyzm.friendly_plans.App;
 import pg.autyzm.friendly_plans.R;
 import pg.autyzm.friendly_plans.databinding.FragmentPlanTaskListBinding;
+import pg.autyzm.friendly_plans.item_touch_helper.SimpleItemTouchHelperCallback;
+import pg.autyzm.friendly_plans.manager_app.view.main_screen.MainActivity;
 import pg.autyzm.friendly_plans.manager_app.view.plan_create_add_tasks.AddTasksToPlanFragment;
-import pg.autyzm.friendly_plans.manager_app.view.task_list.TaskRecyclerViewAdapter;
+import pg.autyzm.friendly_plans.notifications.ToastUserNotifier;
 
 public class PlanTaskListFragment extends Fragment implements PlanTaskListEvents {
 
-    private TaskRecyclerViewAdapter taskListAdapter;
+    private PlanCreateTaskListRecyclerViewAdapter taskListAdapter;
     private Long planId;
 
     @Inject
     PlanTemplateRepository planTemplateRepository;
+    @Inject
+    ToastUserNotifier toastUserNotifier;
+
+    PlanCreateTaskListRecyclerViewAdapter.TaskItemClickListener taskItemClickListener =
+            new PlanCreateTaskListRecyclerViewAdapter.TaskItemClickListener() {
+
+                private boolean removedTask = false;
 
-    TaskRecyclerViewAdapter.TaskItemClickListener taskItemClickListener =
-            new TaskRecyclerViewAdapter.TaskItemClickListener() {
                 @Override
-                public void onTaskItemClick(int position) {
-                    /* What to do after click? Remove task? Edit? */
+                public void onRemoveTaskClick(long itemId) {
+                    planTemplateRepository.removeTaskFromPlan(planId, itemId);
+                    PlanTemplate planTemplate = planTemplateRepository.get(planId);
+                    taskListAdapter
+                            .setTaskItems(planTemplate.getPlanTaskTemplates());
+                    toastUserNotifier.displayNotifications(
+                            R.string.step_removed_message,
+                            getActivity().getApplicationContext());
+                    removedTask = true;
                 }
 
                 @Override
-                public void onRemoveTaskClick(int position){
-                    /*Item remove TODO*/
+                public void onMoveItem() {
+                    Boolean reordered = false;
+                    for(int i = 0; i < taskListAdapter.getItemCount(); i++){
+                        PlanTaskTemplate planTaskItem =  taskListAdapter.getPlanTaskTemplate(i);
+                        if(i != planTaskItem.getOrder()) {
+                            planTemplateRepository.removeTaskFromPlan(planId, planTaskItem.getId());
```

Comment:
The problem could be here, why removing and and creating new, instead of updating the order? Ids will change. And the items in adapter's list have old ids still. I think setting new list to adapter should be added here.